ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/HttpJsonConnectorExecutor.java

```diff
@@ -61,83 +56,50 @@ public class HttpJsonConnectorExecutor implements RemoteConnectorExecutor {
     @Getter
     private Client client;
 
+    private SdkAsyncHttpClient httpClient;
+
     public HttpJsonConnectorExecutor(Connector connector) {
         this.connector = (HttpConnector) connector;
+        this.httpClient = MLHttpClientFactory.getAsyncHttpClient();
     }
 
     @Override
-    public void invokeRemoteModel(MLInput mlInput, Map<String, String> parameters, String payload, List<ModelTensors> tensorOutputs) {
+    public void invokeRemoteModel(
+        MLInput mlInput,
+        Map<String, String> parameters,
+        String payload,
+        Map<Integer, ModelTensors> tensorOutputs,
+        WrappedCountDownLatch countDownLatch,
+        ActionListener<List<ModelTensors>> actionListener
+    ) {
         try {
-            AtomicReference<String> responseRef = new AtomicReference<>("");
-            AtomicReference<Integer> statusCodeRef = new AtomicReference<>();
-
-            HttpUriRequest request;
+            SdkHttpFullRequest request;
```

Comment:
I though this is protocol is for general HTTP, why we are using AWS SDK request?

Reply:
Previously we have separate code for aws and general http/https, but in fact they can be merged together, aws sdk request only have different headers with general one's, so we can use aws sdk request for general ones as well.